backend/src/lib/factories/dataset-factory.ts

```diff
@@ -21,21 +22,41 @@ function createNew(info: DatasetInfo): Dataset {
       raw: info.s3Key.raw,
       json: info.s3Key.json,
     },
+    sourceType: info.sourceType,
   };
 }
 
+function fromItem(item: DatasetItem): Dataset {
+  const id = item.pk.substring(8);
+  let dataset: Dataset = {
+    id,
+    fileName: item.fileName,
+    createdBy: item.createdBy,
+    s3Key: item.s3Key,
+    sourceType: item.sourceType as SourceType,
```

Comment:
what about existing datasets that don't have a `sourceType` defined? We should probably default to `SourceType.fileUpload` here.